src/classes/DOMMatrix.js

```diff
@@ -1,3 +1,23 @@
+function sumMultipleOfMatricesCells(matrix1, matrix2, { i, j }) {
+  let sum = 0;
+  for (let k = 1; k <= 4; k++) {
+    sum += matrix1[`m${k}${j}`] * matrix2[`m${i}${k}`];
+  }
+  return sum;
+}
+
+function multiplyMatrices(leftMatrix, rightMatrix) {
+  for (let i = 1; i <= 4; i++) {
+    for (let j = 1; j <= 4; j++) {
+      leftMatrix[`m${i}${j}`] = sumMultipleOfMatricesCells(
+        leftMatrix,
+        rightMatrix,
+        { i, j }
+      );
+    }
+  }
+}
+
 export default class DOMMatrix {
```

Comment:
I want to reiterate that I appreciate all this hard work! I haven't felt the need to improve this software in a very long time, so when I make this suggestion, it's because I would like to give you the opportunity to make some much-needed changes. I believe this particular class, `DOMMatrix` could be brought up to spec.



Take the following example:

```ts
let a = new DOMMatrix()
a.m11 = "test"
console.log(a.m11)
// NaN
```

So this is going to probably be a lot more work, but currently, are the properties are members of the `DOMMatrix` object. This is not spec-compliant.

It is my recommendation that we create `get` and `set` properties that operate over a backing `Float32Array` buffer.

```ts
    sum += matrix1[`m${k}${j}`] * matrix2[`m${i}${k}`];
```

This particular line of code is very high throughput too, and we should optimize it to use the backing `Float32Array` so that it doesn't require hash lookups to obtain the value of the matrix cell.

Reply:
Can we do the other specs adjustments in a different PR? 
I don't mind doing that. Quite enjoy this. Haven't meddled with matrices since my Webgl days back in 2016 😀.
Regarding the lookup - is it critical for this PR or can we create an issue for this?
It's a mock system evetually

Reply:
> Regarding the lookup - is it critical for this PR or can we create an issue for this?

I think that's a great idea! Feel free to open another pull request that refactors the way `DOMMatrix` works. I can approve this change for now.

Reply:
BTW - I've created the example of the two. Using the string literals is much faster: https://jsbench.me/1vkrz95tvr/1

Reply:
https://jsbench.me/kdkrzb9n63/1

Please take a look at this particular benchmark. Because the float32array would be "source" of the values, already created, that should not be included as part of the test. As expected, using a TypedArray results in a 99% speedup.

Reply:
Of course. What you did is equivalent to preallocation, which is [the solution for GC performance issues](https://yonatankra.com/memory-allocation-and-garbage-collection-in-javascript/). The problem is, that it's not implemented this way in the browser.
![image](https://user-images.githubusercontent.com/6459899/128446455-12596529-f354-4fb1-9210-95aef47eb40d.png)
DOMMatrix doesn't keep an array - it generates a new one every time the method `toFloat64Array` (or 32) is called. Hence preallocating the way you did is not valid because if we change the matrix (e.g. via `scaleSelf`) the array held in memory becomes irrelevant.
Looks like micro optimization at this point, since we need to vary from the actual implementation in the browser. We will need to keep an array at all times and pass its reference (hence making it mutable, which deviates from the spec IMO).
What do you think?

Reply:
Ok, now looking from laptop I see what you mean. Should change the interface of the function to receive the array from the translate/scale methods.